instrumentation/jedis/jedis-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v4_0/JedisConnectionInstrumentation.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jedis.v4_0;
+
+import static io.opentelemetry.javaagent.instrumentation.api.Java8BytecodeBridge.currentContext;
+import static io.opentelemetry.javaagent.instrumentation.jedis.v4_0.JedisSingletons.instrumenter;
+import static java.util.Arrays.asList;
+import static net.bytebuddy.matcher.ElementMatchers.is;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import java.net.Socket;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import redis.clients.jedis.CommandArguments;
+import redis.clients.jedis.commands.ProtocolCommand;
+
+public class JedisConnectionInstrumentation implements TypeInstrumentation {
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("redis.clients.jedis.Connection");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        isMethod()
+            .and(named("sendCommand"))
+            .and(takesArguments(1))
+            .and(takesArgument(0, named("redis.clients.jedis.CommandArguments"))),
+        this.getClass().getName() + "$SendCommand2Advice");
+
+    transformer.applyAdviceToMethod(
+        isMethod()
+            .and(named("sendCommand"))
+            .and(takesArguments(2))
+            .and(takesArgument(0, named("redis.clients.jedis.commands.ProtocolCommand")))
+            .and(takesArgument(1, is(byte[][].class))),
+        this.getClass().getName() + "$SendCommandAdvice");
+    // FIXME: This instrumentation only incorporates sending the command, not processing the result.
```

Comment:
@laurit Please create an issue about this gap

Reply:
@iNikem this is a copy paste from previous version